packages/next/src/build/analysis/get-page-static-info.ts

```diff
@@ -231,13 +231,17 @@ function getMiddlewareConfig(
   return result
 }
 
-let warnedAboutExperimentalEdge = false
-function warnAboutExperimentalEdge() {
-  if (warnedAboutExperimentalEdge) {
+let apiRouteWarnings = new Set()
```

Comment:
```suggestion
const apiRouteWarnings = new Set()
```

Reply:
This seems to cause a small memory leak because the list is not cleaned up at some point

Reply:
Thanks for the review!

You're right about the memory leak in Dev mode.  I can't think of an "event" that we could listen to to clear the set. Clearing the set means displaying warnings for files we already warned about in the past.

I hope we can allow it because the set is only growing when users are changing the runtime key of their new files. 

In other words, users would need to create a massive amount of new files containing `experimental-edge` to fill up this set. Which hopefully should not happen.

Reply:
Maybe it should use the lru-cache package with a set limit?

Reply:
Very good call! I've used LRU with a max number of 250 entries. This shields us from memory leaks.
Hopefully no one is actively developing with 250 API routes using `experimental-edge`.